File: sdk/node.js/core/EasyReconTemplate.js
/**
 * 对账SDK核心模板类
 */
class EasyReconTemplate {
  /**
   * 构造函数
   * @param {Object} realtimeReconService - 实时对账服务
   * @param {Object} timingReconService - 定时对账服务
   */
  constructor(realtimeReconService, timingReconService) {
    this.realtimeReconService = realtimeReconService;
    this.timingReconService = timingReconService;
  }

  /**
   * 执行实时对账
   * @param {Object} orderMain - 订单主记录
   * @param {Array} splitSubs - 分账子记录列表
   * @returns {Promise<boolean>} 对账结果
   */
  async doRealtimeRecon(orderMain, splitSubs) {
    return this.realtimeReconService.doRealtimeRecon(orderMain, splitSubs);
  }

  /**
   * 执行定时对账
   * @param {string} dateStr - 对账日期（YYYY-MM-DD）
   * @returns {Promise<boolean>} 对账结果
   */
  async doTimingRecon(dateStr) {
    return this.timingReconService.doTimingRecon(dateStr);
  }
}

module.exports = EasyReconTemplate;

File: sdk/python/core/easy_recon_template.py
from typing import List
from entity.recon_order_main import ReconOrderMain
from entity.recon_order_split_sub import ReconOrderSplitSub
from service.realtime_recon_service import RealtimeReconService
from service.timing_recon_service import TimingReconService


class EasyReconTemplate:
    """对账SDK核心模板类"""

    def __init__(self, realtime_recon_service: RealtimeReconService, timing_recon_service: TimingReconService):
        self.realtime_recon_service = realtime_recon_service
        self.timing_recon_service = timing_recon_service

    def do_realtime_recon(self, order_main: ReconOrderMain, split_subs: List[ReconOrderSplitSub]) -> bool:
        """执行实时对账"""
        return self.realtime_recon_service.do_realtime_recon(order_main, split_subs)

    def do_timing_recon(self, date_str: str) -> bool:
        """执行定时对账"""
        return self.timing_recon_service.do_timing_recon(date_str)
